### ISSUE TITLE
Better error messages when sops can't obtain a data key for decryption

### ISSUE THREAD

**Issue by aywrite** [isindir/sops-secrets-operator#23]
Currently if the sops-operator attempts to decrypt a secret encrypted via KMS when it does not have sufficient IAM privileges to assume the role required for decryption it will output a message like:

```json
{"error": "Error getting data key: 0 successful groups required, got 0"}
```

This error comes from the error output by the sops `GetDataKey` method. When the same error is hit using the sops CLI, extra information is added by sops by using the `UserError` method. By performing a type assertion it should be possible to add the same additional information to help with debugging.

A minimal example would be to just re-use the `UserError` method (although this is optimized for CLI output not structured logging).
```go
key, err := tree.Metadata.GetDataKey()
if userErr, ok := err.(sops.UserError); ok {
      err = fmt.Errorf(userErr.UserError())
}
```
This produces an error message like:
```json
{"error": "Failed to get the data key required to decrypt the SOPS file.\n\nGroup 0: FAILED\n  arn:aws:kms:xxxxxxx FAILED\n    - | Error decrypting key: WebIdentityErr: failed to retrieve\n      | credentials\n      | caused by: AccessDenied: Not authorized to perform\n      | sts:AssumeRoleWithWebIdentity\n      | \tstatus code: 403, request id:\n      | bxxxxx \n\nRecovery failed because no master key was able to decrypt the file. In\norder for SOPS to recover the file, at least one key has to be successful,\nbut none were."}
```

**Comment by isindir** [comment]
I'll look at that

**Comment by isindir** [comment]
Should be fixed now, see #25
